### PR TITLE
Add ResourceBuilder support to plugin

### LINKED ISSUE #2028: Update plugin to support Resource Detector
Support OpenTelemetry Resource Detector - https://github.com/open-telemetry/opentelemetry-dotnet/blob/bb0c381478a5d6b6a3dd52cc9990d3582e908767/docs/trace/extending-the-sdk/README.md?plain=1#L340

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
## Why

Fixes #2028 

## What

This will help enable the Resource Detectors through plugins.

## Checklist

- [X] `CHANGELOG.md` is updated.
- [X] Documentation is updated.
- [X] New features are covered by tests.

### CODE REVIEW

**Comment by Kielek** [comment]
I am not sure if you really need it. Please consider this https://github.com/signalfx/splunk-otel-dotnet/blob/488085436a1faab8a61a15a8184f527e20d12c8a/src/Splunk.OpenTelemetry.AutoInstrumentation/Traces.cs#L49-L52

**Comment by rajkumar-rangaraj** [comment]
> I am not sure if you really need it. Please consider this https://github.com/signalfx/splunk-otel-dotnet/blob/488085436a1faab8a61a15a8184f527e20d12c8a/src/Splunk.OpenTelemetry.AutoInstrumentation/Traces.cs#L49-L52
> 
> It requires 3 independent plugins: for Traces, Logs and Metrics, but in general it works fine for Splunk distribution.

We don't have an option for user to integrate resource detector with auto-instrumentation. If some one wants to send data to OpenTelemetry Collector and use detector, proposed changes will enable easier integration than workaround to add with other signal builders.

File: docs/plugins.md
Comment (Kielek) [inline docs/plugins.md:65]: Based on current discussion.
I would put some more information here that it will be called independently for each signal without possibility to verify which signal is really configured.
What is more settings from this method can be overriden by ConfigureTracesOptions/ConfigureMeterProvider/ConfigureLogsOptions by calling SetResourceBuilder with new object.

File: src/OpenTelemetry.AutoInstrumentation/Configuration/PluginsConfigurationHelper.cs
Comment (pjanotti) [inline src/OpenTelemetry.AutoInstrumentation/Configuration/PluginsConfigurationHelper.cs:42]: nit: no need for an else branch :)

File: test/OpenTelemetry.AutoInstrumentation.Tests/Configuration/PluginManagerTests.cs
Comment (pjanotti) [inline test/OpenTelemetry.AutoInstrumentation.Tests/Configuration/PluginManagerTests.cs:205]: Shouldn't be a check for this attribute somewhere in the test?

File: test/OpenTelemetry.AutoInstrumentation.Tests/Configurations/PluginManagerTests.cs
Comment (rajkumar-rangaraj) [inline test/OpenTelemetry.AutoInstrumentation.Tests/Configurations/PluginManagerTests.cs:168]: @pellared Need your expertise here. Want to check of we could combine this test with `InvokeProviderPluginSuccess`? Tried created a ResourceBuilder mock, as the `AddAttribute` method takes `List<KeyValuePair>` verifiable was not happy with it. Hence created a new test, want to ensure I'm not missing anything here.

**Review by pjanotti — APPROVED** [review]
LGTM @rajkumar-rangaraj I will let you decide if you want to wait for @pellared's reply before merging it.

File: .cspell/other.txt
Comment (Kielek) [inline .cspell/other.txt:36]:
```suggestion
vcpkg
```

File: docs/plugins.md
Comment (Kielek) [inline docs/plugins.md:67]:
```suggestion
        // This method could be overridden by ConfigureTracesOptions, ConfigureMeterProvider and ConfigureLogsOptions
```

**Review by Kielek — APPROVED** [review]

File: test/OpenTelemetry.AutoInstrumentation.Tests/Configurations/PluginManagerTests.cs
Comment (pellared) [reply test/OpenTelemetry.AutoInstrumentation.Tests/Configurations/PluginManagerTests.cs:168]: I prefer this test test a lot more than `InvokeProviderPluginSuccess`. It does not use mocks, tests against "real" objects, therefore gives more trustworthiness. Moreover, for me it is easier to read.

However, personally, I would add a test to `IntegrationTests.PluginsTests` that would check the new functionality. You can add use in functionality to `TestApplication.Plugins.Plugin` and use the `collector.ResourceExpector` for asserting.

Comment (rajkumar-rangaraj) [reply test/OpenTelemetry.AutoInstrumentation.Tests/Configurations/PluginManagerTests.cs:168]: Adding this to issue #1742 for adding integration test for this scenario.

**Pull request merged**
